### PR TITLE
Improve robustness of E2E Golang tests

### PR DESCRIPTION
**Pull request by iblancasa**
## Which problem is this PR solving?
- Improve robustness of E2E Golang tests

### CODE REVIEW

File: test/e2e/all_in_one_test.go [hunk 1]
```diff
@@ -95,38 +92,11 @@ func (suite *AllInOneTestSuite) TestAllInOneWithIngress() {
...
 
-		for _, v := range resp.Data {
-			if v == "jaeger-query" {
-				return true, nil
-			}
-		}
-
-		return false, nil
-	})
+	// We just need to check there ere not errors in the HTTP response and the REST API is available
```
Comment (rubenvp8510) [inline test/e2e/all_in_one_test.go:99]:
```suggestion
	// We just need to check there are not errors in the HTTP response and the REST API is available
```

File: test/e2e/wait_util.go [hunk 2]
```diff
@@ -232,3 +237,82 @@ func WaitForSecret(secretName, secretNamespace string) {
...
+
+	err = wait.Poll(retryInterval, timeout*2, func() (done bool, err error) {
+		res, err := httpClient.Do(req)
+
+		if res != nil {
+			logrus.Infof("response status code: %d", res.StatusCode)
+		}
+
+		if err != nil {
+			if strings.Contains(err.Error(), "Timeout exceeded") {
```
Comment (rubenvp8510) [inline test/e2e/wait_util.go:262]: I think there is a better way to do this, checking the error type
You can do:

```
if e,ok := err.(net.Error); ok && e.Timeout() {
    // This was a timeout
} else if err != nil {
    // This was an error, but not a timeout
}
```
https://pkg.go.dev/net#Error

File: test/e2e/all_in_one_test.go (hunk 1 above)
Comment (iblancasa) [reply test/e2e/all_in_one_test.go:99]: Good catch!

File: test/e2e/wait_util.go (hunk 2 above)
Comment (iblancasa) [reply test/e2e/wait_util.go:262]: Oh, really interesting. I'll apply this. Thanks!

**Review by rubenvp8510 — APPROVED** [review]

**Pull request merged**